### PR TITLE
added two parameters: only-get-or-head and check-index

### PR DESCRIPTION
**Pull request by tts-sdrissen**
## Description
I am using your project to mock an API for testing purposes.
I needed two more functions to make it work.

## Problem
I need to test API requests to something like /users/1 AND /users/1/profile .
So what I did was creating a folder structure and placed in the "/users/1"-folder an index.html
But it returns a directory listing even if a index.html is present.

And I need POST requests, but the original code blocks them.

## Solution
Introduced parameter check-index for looking for index.html, before return directory listing.
Introduced parameter only-get-or-head to allow to set it to false, to ignore method check.

## Notes
If you merge this, nothing significant will happen for existing users, because the default parameters say not to use the functions introduced in this request.

### CODE REVIEW

**Comment by Eun** [comment]
Thank you for your contribution! :+1: 
I see the usecase.
I also would like to see the feature implemented, however I think there are some problems with long time use of the features you added.

1. When introducing an index handler it is required that the index file can be specified. (Some users use `index.html`, others use `index.htm`)  
  The best solution would be to add an array parameter where devs could specify what should be handled:
  ```index-files: [index.html, index.htm]```
1.1. Making the directory listing configurable would be the next logical step when adding an index handler.
3. The "only get or head" feature should follow the same pattern:
  ```allowed-methods: [GET, HEAD]```

Let me know if you intend to make the changes. :smiley:

**Comment by tts-sdrissen** [comment]
Nice to see, that you reviewed my pull request!
I think that your comments are reasonable and I started to make the changes you suggested.

I committed "requested changes" and I can see it in this pull request - not sure, if you can see it, too.
But it is not tested yet... I will approve it, as soon as my Github Action for another projects ran without an issue.

**Comment by tts-sdrissen** [comment]
Okay, the requested modifications are working.
And you should see the commit "requested changes" and "minor fixes".

Be aware, that I modified the README in "my" repo for copyright reasons.
Better check every file before merge 😅

File: server.js [hunk 1]
```diff
@@ -60,38 +64,53 @@ function deploy(config, ready) {
...
+				}
+
+				for (const file of fs.readdirSync(requestedFile)) {
+					const fullPath = requestedFile + file;
+					response.write(`<a href="/${toPosixPath(fullPath.slice(cwd.length))}">${file}</a>\n`);
+				}
+				response.write('</pre>');
+				response.end();
+				return;
+			}
```
Comment (Eun) [inline server.js:113]: 1. Removed idention
2. utilized the `every` function result (https://jsfiddle.net/dnzmtfhu/)
3. changed
    ```
    let indexFile = requestedFile + "index.html";
    ```
    to
    ```
    const indexFile = requestedFile + elem;
    ```
 
```suggestion
	if (!requestedFile.endsWith(path.sep)) {
		requestedFile += path.sep;
	}
	const noIndexFound = config.indexFiles.every(elem => {
		const indexFile = requestedFile + elem;
		if(fs.existsSync(indexFile)){
			requestedFile = indexFile;
			stat = fs.statSync(requestedFile);
			return false;
		}
		return true;
	});

	if(noIndexFound) {
		response.writeHead(200, 'OK', { 'Content-Type': 'text/html' });

		if (request.method === 'HEAD') {
			response.end();
			return;
		}
		response.write('<pre>\n');

		let parentDir = path.resolve(path.normalize(path.join(requestedFile, '..')));
		if (!parentDir.endsWith(path.sep)) {
			parentDir += path.sep;
		}
		if (parentDir.startsWith(cwd)) {
			let parentLink = '/' + toPosixPath(parentDir.slice(cwd.length));
			if (parentLink === '/.') {
				parentLink = '/';
			}
			response.write(`<a href="${parentLink}">..</a>\n`);
		}

		for (const file of fs.readdirSync(requestedFile)) {
			const fullPath = requestedFile + file;
			response.write(`<a href="/${toPosixPath(fullPath.slice(cwd.length))}">${file}</a>\n`);
		}
		response.write('</pre>');
		response.end();
		return;
	}
```

File: action.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # https://help.github.com/en/articles/metadata-syntax-for-github-actions
-name: 'HTTP Server Action'
+name: 'HTTP Server Action (tts-mod)'
```
Comment (Eun) [inline action.yml:2]:
```suggestion
name: 'HTTP Server Action'
```

File: README.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # http-server-action
 ---
-An action that spawns an http server to serve files.
+An action that spawns an http server to serve files. 
+This project is forked from https://github.com/Eun/http-server-action
+
+Changes: Improved to handle index.html in root of directories.
```
Comment (Eun) [inline README.md:6]:
```suggestion
```

**Review by Eun — CHANGES_REQUESTED** [review]
Thanks, for the work! :muscle: 
I checked the code, just one tiny bug, a bit of cleaning up. Then I would love to merge :smiley: 
I suggested all the changes so you just need to accept them :+1:

**Comment by tts-sdrissen** [comment]
I approved the changes and found an error in the README.md - Tabs instead of spaces.
After you merge those changes, can I delete my fork?

**Comment by Eun** [comment]
Yes you can delete your fork, thanks again :)

**Comment by tts-sdrissen** [comment]
It even says "If you wish, you can delete this fork of Eun/http-server-action in the settings."

It was a pleasure to collaborate with you.

**Pull request merged**